### DISCUSSION TITLE
Wildcard when making yup schema

### DISCUSSION THREAD

**Discussion by FilipBrannlund** [react-hook-form/react-hook-form#5308] [Q&A] [answered]
Hello!

Is there any way when making the validation schema to specify name with wildcard?
I would like to create a general rule for all firstName-fields.

```
const persons = [
    { firstName: "", lastName: "" },
    { firstName: "", lastName: "" },
    { firstName: "", lastName: "" },
    { firstName: "", lastName: "" }
  ];
const schema = yup.object().shape({
    firstName: yup.string().required()
  });
```

I would like to do something like:
```
const schema = yup.object().shape({
    person*firstName yup.string().required()
  });
```

https://codesandbox.io/s/admiring-bhaskara-rgtho?file=/src/App.js

**Comment by bluebill1049** [comment, marked as answer]
I would post this question at yup, I think you will get a better and accurate answer.

**Reply by FilipBrannlund** [reply]
Thanks, I will.
I was thinking, what is the best practise when having an array with object that you want to iterate out as inputs.
Basically just an map-function and displaying propeties from the object as inputs or useFieldArray?

My first impression about useFieldArray is to use it when handling dynamic form that hasn't any prefill data instead use it when starting from an index of 0.

**Reply by bluebill1049** [reply]
I would use useFieldArray for that.

**Reply by FilipBrannlund** [reply]
Okey, do you know if there is any CSB where an object's properties is getting iterated to inputs with usefieldarray. Currently I can only find CBS's where it starts from an empty object.

E.g there is an array of 5 person object as init, first iterate out all persons and then be able to append 6,7,8 more persons using usefieldarray.
